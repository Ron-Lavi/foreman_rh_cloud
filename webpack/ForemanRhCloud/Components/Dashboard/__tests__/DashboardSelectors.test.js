import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import {
  selectDashboard,
  selectPollingProcessID,
  selectActiveTab,
  selectUploading,
  selectGenerating,
} from '../DashboardSelectors';
import {
  logs,
  completed,
  pollingProcessID,
  activeTab,
  accountID,
} from '../Dashboard.fixtures';

const state = {
  dashboard: {
    [accountID]: {
      generating: {
        logs,
        completed,
      },
      uploading: {
        logs,
        completed,
      },
      activeTab,
      pollingProcessID,
    },
  },
};

const fixtures = {
  'should return Dashboard': () => selectDashboard(state, accountID),
  'should return Dashboard uploading': () => selectUploading(state, accountID),
  'should return Dashboard generating': () =>
    selectGenerating(state, accountID),
  'should return Dashboard pollingProcessID': () =>
    selectPollingProcessID(state, accountID),
  'should return Dashboard activeTab': () => selectActiveTab(state, accountID),
};

describe('Dashboard selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
