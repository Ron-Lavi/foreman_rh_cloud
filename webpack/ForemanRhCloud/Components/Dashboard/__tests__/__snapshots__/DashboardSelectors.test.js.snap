// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard selectors should return Dashboard 1`] = `
Object {
  "activeTab": "uploads",
  "generating": Object {
    "completed": 25,
    "logs": Array [
      "some-logs...",
    ],
  },
  "pollingProcessID": 1,
  "uploading": Object {
    "completed": 25,
    "logs": Array [
      "some-logs...",
    ],
  },
}
`;

exports[`Dashboard selectors should return Dashboard activeTab 1`] = `"uploads"`;

exports[`Dashboard selectors should return Dashboard generating 1`] = `
Object {
  "completed": 25,
  "logs": Array [
    "some-logs...",
  ],
}
`;

exports[`Dashboard selectors should return Dashboard pollingProcessID 1`] = `1`;

exports[`Dashboard selectors should return Dashboard uploading 1`] = `
Object {
  "completed": 25,
  "logs": Array [
    "some-logs...",
  ],
}
`;
